sagemaker-featurestore/feature_store_kms_key_encryption.ipynb

```diff
@@ -0,0 +1,560 @@
+    "customers_feature_group.create(\n",
+    "    s3_uri=f\"s3://{s3_bucket_name}/{prefix}\",\n",
+    "    record_identifier_name=record_identifier_feature_name,\n",
+    "    event_time_feature_name=\"EventTime\",\n",
+    "    role_arn=role,\n",
+    "    enable_online_store=False, \n",
+    "    offline_store_kms_key_id = 'arn:aws:kms:us-east-1:123456789012:key/'+ new_kms_key['KeyMetadata']['KeyId']\n",
+    ")\n",
+    "\n",
+    "orders_feature_group.create(\n",
+    "    s3_uri=f\"s3://{s3_bucket_name}/{prefix}\",\n",
+    "    record_identifier_name=record_identifier_feature_name,\n",
+    "    event_time_feature_name=\"EventTime\",\n",
+    "    role_arn=role,\n",
+    "    enable_online_store=False,\n",
+    "    offline_store_kms_key_id = 'arn:aws:kms:us-east-1:123456789012:key/'+new_kms_key['KeyMetadata']['KeyId']\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## How to verify that your KMS key is being used to encrypt your data in your Online or Offline Feature Store? \n",
+    "\n",
+    "### Online Store Verification\n",
+    "To demonstrate that your data is being encrypted in your Online store, use your `kms` client from `boto3` to list the grants under your KMS key. It should show 'SageMakerFeatureStore-' and the name of your feature group you created and should list these operations under Operations:`['Decrypt','Encrypt','GenerateDataKey','ReEncryptFrom','ReEncryptTo','CreateGrant','RetireGrant','DescribeKey']`\n",
+    "\n",
+    "An alternative way for you to check that your data is encrypted in your Online store is to check [Cloud Trails](https://console.aws.amazon.com/cloudtrail/) and navigate to your account name. Once here, under General details you should see that SSE-KMS encryption is enabled and with your AWS KMS key shown below it. Below is a screenshot showing this: \n",
+    "\n",
+    "![Cloud Trails](images/cloud-trails.png)\n",
+    "\n",
+    "### Offline Store Verification\n",
+    "To verify that your data in being encrypted in your Offline store, you must navigate to your S3 bucket through the [Console](https://console.aws.amazon.com/s3/home?region=us-east-1) and then navigate to your prefix, offline store, feature group name and into the /data/ folder. Once here, select a parquet file which is the file containing your feature group data. For this example, the directory path in S3 was this: \n",
+    "\n",
+    "`Amazon S3/MYBUCKET/PREFIX/123456789012/sagemaker/region/offline-store/customers-feature-group-23-22-44-47/data/year=2021/month=03/day=23/hour=22/20210323T224448Z_IdfObJjhpqLQ5rmG.parquet.` \n",
+    "\n",
+    "After selecting the parquet file, navigate to Server-side encryption settings. It should mention that Default encryption is enabled and reference (SSE-KMS) under server-side encryption. If this show, then your data is being encrypted in the offline store. Below is a screenshot of how this should look like in the console: \n",
```

Comment:
"After you select the Parquet file, navigate to the server-side encryption settings. If default encryption is enabled and it references (SSE-KMS) under server-side encryption, your data is being encrypted in the offline store, as shown in the following screenshot."